164796849 Password Reset Functionality

Simplemart17 opened the pull request:
#### Description
Allows users to request for password reset
#### Type of change
- [ ] Bug Fix (a non-breaking change which fixes an issue)
- [x] New Feature (a non-breaking change that adds functionality)
- [ ] New Chore (a non-breaking change that adds functionality)
- [ ] Breaking Change (a fix or feature which would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update
#### How Has This Been Tested?
- [ ] End to End
- [x] Integration
#### Checklist:
- [x] Unit test
- [ ] Code coverage
#### Thought Process
- Add form component
- Connect to redux state
- Test page component
- Test actions, reducer
#### Screen Shot
<img width="1440" alt="Screenshot 2019-05-14 at 6 27 51 PM" src="https://user-images.githubusercontent.com/41796727/57718973-9a646500-7676-11e9-8559-69e42198644f.png">
<img width="1414" alt="Screenshot 2019-05-18 at 8 47 06 PM" src="https://user-images.githubusercontent.com/41796727/57974359-c0f40a00-79ae-11e9-91df-b3ec879e331e.png">
<img width="854" alt="Screenshot 2019-05-18 at 8 47 39 PM" src="https://user-images.githubusercontent.com/41796727/57974361-cc473580-79ae-11e9-9e1e-13947fee7f33.png">
<img width="760" alt="Screenshot 2019-05-18 at 8 48 43 PM" src="https://user-images.githubusercontent.com/41796727/57974365-d8cb8e00-79ae-11e9-86f6-09a58f4360c2.png">

#### PT
[#164796849](https://www.pivotaltracker.com/story/show/164796849)

Simplemart17 committed:
add function for password reset

Simplemart17 committed:
164796849-feature(forgotPassword): add functionality to forgot password

- functionality to forgot password feature

[Delivers #164796849]

Simplemart17 committed:
164796849-feature(forgotPassword): implement forgot password functionality

- add functionality for forgot password

[Delivers #164796849]

Simplemart17 committed:
164796849-feature(ForgotPassword): add test to cover more lines

- add tests for actions, reducer

[Delivers #164796849]

micah-akpan commented on src/lib/forgotPassword.js:
```diff
@@ -0,0 +1,6 @@
+export const userForgotPassword = async email => {
+  const forgotPassword = await axios.post('/forgot-password', { email });
+  return forgotPassword;
+};
```
Can you use a try/catch here, since you are doing some asynchronous operations here?

micah-akpan reviewed (commented):
@Simplemart17 Nice work. Can you make the suggested changes, so I can approve it?

Simplemart17 committed:
164796849-feature(resetPassword): functionality for reseting password

- function for user to reset password

[Delivers #164796849]

Simplemart17 committed:
--amend

Simplemart17 committed:
164796849-feature(resetPassword): add test

- add test for the reset password functionality

[Delivers #164796849]

Simplemart17 committed:
--amend

Simplemart17 committed:
164796849-feature(passwordReset): password reset functionality

- add functionality for forgot password and reset password

[Delivers #164796849]

ssanusi commented on src/pages/ResetPassword.jsx:
```diff
@@ -0,0 +1,112 @@
+    const errors = {};
+    if (!/^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\\$%\\^&\\*])(?=.{8,})/.test(data.password)) {
+      errors.password = 'Password should be At least a lowercase character, uppercase character, numeric character, special character and eight characters long.';
+    }
```
@Simplemart17 nice work I think you might want to put all your validation in util folder

Simplemart17 replied:
Thank you for the feedback, I don't think it might be necessary considering the fact that it is a line of code.

Simplemart17 committed:
--amend

164796849-feature(resetPassword): update pull request

- update pull request

[Delivers #164796849]

chikeozulumba reviewed (dismissed):
@micah-akpan @Simplemart17 LGTM
